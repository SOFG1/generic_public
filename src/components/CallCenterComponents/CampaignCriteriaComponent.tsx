import React, { useCallback, useEffect, useMemo, useState } from "react";
import styled from "styled-components";
import { handle } from "../../api";
import { GoogleAds } from "../../api/googleAds";
import { RawData } from "../../api/rawData";
import { useAppActions } from "../../store/app";
import { useGoogleAdsState } from "../../store/googleAds";
import { useUserState } from "../../store/user";
import { desktopBp } from "../../styles/variables";
import { Button } from "../../UI/Button";
import {
  Dropdown,
  DropdownSearchFetch,
  DropdownWithSearch,
} from "../../UI/Dropdown";
import { Input } from "../../UI/Input";
import { HintText, StyledHint } from "../../UI/InputHint/InputHint";
import { Loader } from "../../UI/Spinners";
import { InfoIcon } from "../../UI/Svg";
import { isoCodes } from "../../utils/isoCodes";
import { Title } from "../common/Title";
import { activityList } from "../../config/userActivityList";
import { useTranslation } from "react-i18next";

const StyledTitle = styled(Title)`
  margin-top: 0;
  margin-bottom: 0.52vw;
  @media screen and (max-width: ${desktopBp}) {
    margin-bottom: 7px;
  }
`;

const InputsContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1.04vw;
  padding: 0.52vw 9.38vw;
  @media screen and (max-width: ${desktopBp}) {
    gap: 13px;
    padding: 7px 118px;
  }
  @media screen and (max-width: 800px) {
    padding: 10px;
  }
`;

const InputBox = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 0.52vw;
  @media screen and (max-width: ${desktopBp}) {
    gap: 7px;
  }
`;


const ClearButton = styled(Button)`
  padding: 4px 5px;
  font-size: 0.73vw;
  width: 4.69vw;
  align-self: flex-end;
  @media screen and (max-width: ${desktopBp}) {
    font-size: 10px;
    width: 59px;
  }
`;

const StyledAddBtn = styled(Button)` 
  margin: 0 auto;
`;

const StyledLoader = styled(Loader)`
  margin: 0 auto;
`;

const CampaignCriteriaComponent = React.memo(() => {
  const { t } = useTranslation()
  const { token } = useUserState();
  const { campaigns, selectedCustomerId } = useGoogleAdsState();
  const { onShowAlert } = useAppActions();
  const [formData, setFormData] = useState<{ [key: string]: any }>({});
  const [isFetching, setIsFetching] = useState<boolean>(false);
  const [cityOptions, setCityOptions] = useState<
    { item: string; value: string }[]
  >([]);

  const campaignsOptions = useMemo(() => {
    return campaigns.map((c) => ({ item: c.name, value: c.campaign_id }));
  }, [campaigns]);

  const handleAdd = useCallback(async () => {
    const reqData: any = {
      customer_id: String(selectedCustomerId),
      ...formData,
    };
    if (formData.lang_list) {
      reqData.lang_list = [formData.lang_list];
    }
    if (token) {
      setIsFetching(true);
      const [dataRes, dataErr] = await handle(
        GoogleAds.setCampaignCriteria(token, reqData)
      );
      setIsFetching(false);
      if (!dataErr) {
        onShowAlert(true, "Success");
      }
      if (dataErr) {
        onShowAlert(false, dataErr.error);
      }
    }
  }, [token, selectedCustomerId, formData]);

  const handleClear = () => {
    setFormData({});
  };

  const showClearButton = useMemo(() => {
    let isChanged = false;
    Object.values(formData).forEach((data) => {
      if (data) isChanged = true;
    });
    return isChanged;
  }, [formData]);

  const fetchCityOptions = useCallback(
    async (query: string) => {
      if (token) {
        const [dataRes, dataErr] = await handle(
          RawData.getFilterData({ filter_name: "city", option: "query" }, token)
        );
        if (dataRes) {
          const citiesOptions = dataRes.map((c: [number, string]) => ({
            item: c[1],
            value: c[1],
          }));
          setCityOptions(citiesOptions);
        }
        if (dataErr) {
          console.log(dataErr);
        }
      }
    },
    [token]
  );

  useEffect(() => {
    fetchCityOptions("");
  }, [fetchCityOptions]);

  return (
    <>
      <StyledTitle>Create Campaign Criteria</StyledTitle>
      <InputsContainer className="styled_create_campaign_inputs">
        <InputBox>
          <Dropdown
            label="campaign"
            placeholder="campaign"
            onSelect={(v) => setFormData((p) => ({ ...p, campaign_id: v }))}
            value={formData.campaign_id}
            options={campaignsOptions}
          />
          <StyledHint>
            <HintText>{t("google_ads-criteria_campaign-h2")}</HintText>
            <InfoIcon />
          </StyledHint>
        </InputBox>

        <InputBox>
          <Input
            name="locs"
            type="text"
            placeholder="locs"
            label="locs"
            value={formData.locs || ""}
            onChange={(v) => setFormData((p) => ({ ...p, locs: v }))}
          />
          <StyledHint>
            <HintText>{t("google_ads-criteria_campaign-h2")}</HintText>
            <InfoIcon />
          </StyledHint>
        </InputBox>

        <InputBox>
          <DropdownWithSearch
            placeholder="country_code"
            label="country_code"
            options={isoCodes}
            value={formData.country_code || ""}
            onSelect={(v) => setFormData((p) => ({ ...p, country_code: v }))}
          />
          <StyledHint>
            <HintText>{t("google_ads-criteria_campaign-h3")}</HintText>
            <InfoIcon />
          </StyledHint>
        </InputBox>

        <InputBox>
          <DropdownSearchFetch
            options={cityOptions}
            value={formData.city || ""}
            placeholder="city"
            onSelect={(v) => setFormData((p) => ({ ...p, city: v }))}
            onSearch={(query) => fetchCityOptions(query)}
            label="city"
          />
          <StyledHint>
            <HintText>City</HintText>
            <InfoIcon />
          </StyledHint>
        </InputBox>

        <InputBox>
          <Input
            name="locale"
            type="text"
            placeholder="locale"
            label="locale"
            value={formData.locale || ""}
            onChange={(v) => setFormData((p) => ({ ...p, locale: v }))}
          />
          <StyledHint>
            <HintText>{t("google_ads-criteria_campaign-h4")}</HintText>
            <InfoIcon />
          </StyledHint>
        </InputBox>

        {showClearButton && (
          <ClearButton onClick={handleClear}>Clear all</ClearButton>
        )}
      </InputsContainer>
      {isFetching ? (
        <StyledLoader />
      ) : (
        <StyledAddBtn onClick={handleAdd} data-action={activityList["call-center-google-criteria_create"]}>Set a criteria</StyledAddBtn>
      )}
    </>
  );
});

export default CampaignCriteriaComponent;
