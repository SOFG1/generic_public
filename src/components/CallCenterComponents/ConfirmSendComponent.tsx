import React, { useEffect } from "react";
import styled from "styled-components";
import { colors } from "../../styles/colors";
import { desktopBp } from "../../styles/variables";
import { NoIcon, YesIcon } from "../../UI/Svg";
import { Text } from "../common/Text";
import { Title } from "../common/Title";

const ConfirmDelete = styled.div`
  background: #ffffff;
  border: 1px solid ${colors.blue_green};
  box-sizing: border-box;
  box-shadow: 0 4px 15px rgba(0, 0, 0, 0.25);
  padding: 0.52vw 1.56vw;
  border-radius: 14px;
  position: absolute;
  left: 50%;
  transform: translate(-50%, -100%);
  top: 0;
  z-index: 1000;
  &:before {
    content: "";
    position: absolute;
    top: 100%;
    left: 50%;
    transform: translateX(-50%);
    display: block;
    width: 0;
    height: 0;
    border-top: 0.78vw solid ${colors.blue_green};
    border-right: 0.78vw solid transparent;
    border-bottom: 0.78vw solid transparent;
    border-left: 0.78vw solid transparent;
  }
  &:after {
    content: "";
    position: absolute;
    top: 100%;
    left: 50%;
    transform: translateX(-50%);
    display: block;
    width: 0;
    height: 0;
    border-top: 0.73vw solid ${colors.white};
    border-right: 0.73vw solid transparent;
    border-bottom: 0.73vw solid transparent;
    border-left: 0.73vw solid transparent;
  }

  @media screen and (max-width: ${desktopBp}) {
    padding: 7px 20px;
    &:before {
      border-top: 10px solid ${colors.blue_green};
      border-right: 10px solid transparent;
      border-bottom: 10px solid transparent;
      border-left: 10px solid transparent;
    }
    &:after {
      border-top: 9px solid ${colors.white};
      border-right: 9px solid transparent;
      border-bottom: 9px solid transparent;
      border-left: 9px solid transparent;
    }
  }
`;

const ButtonBlock = styled.div`
  display: flex;
  align-content: center;
  align-items: center;
  justify-content: center;
  margin-top: 3px;
`;

const IconButton = styled.div`
  border: none;
  background-color: transparent;
  outline: none;
  margin: 0 0.52vw;
  padding: 0;
  display: flex;
  align-content: center;
  align-items: center;
  transition: all 0.25s ease;
  justify-content: center;
  cursor: pointer;
  &:hover {
    filter: drop-shadow(0 2px 2px rgba(34, 34, 34, 0.7));
  }
  @media screen and (max-width: ${desktopBp}) {
    margin: 0 7px;
  }
`;
const TextStyled = styled(Text)`
  font-size: 0.73vw;
  color: ${colors.graphite_6};
  text-align: center;
  @media screen and (max-width: ${desktopBp}) {
    font-size: 9px;
  }
`;

interface IProps {
  text: string;
  onClose: () => void;
  onConfirm: () => void;
}

const ConfirmSendComponent = React.memo(
  ({ onConfirm, onClose, text }: IProps) => {
    const handleClose = (e: React.MouseEvent) => {
      e.stopPropagation();
      onClose();
    };

    const handleDismiss = (e: any) => {
      if (!e.target.closest(".confirm_delete")) onClose();
    };

    useEffect(() => {
      document.addEventListener("click", handleDismiss);
      return () => document.removeEventListener("click", handleDismiss);
    }, []);

    const handleConfirm = (e: any) => {
      e.stopPropagation();
      onConfirm();
      onClose();
    };

    return (
      <ConfirmDelete className="confirm_delete">
        <TextStyled>{text}</TextStyled>
        <ButtonBlock>
          <IconButton onClick={handleConfirm}>
            <YesIcon />
          </IconButton>
          <IconButton onClick={handleClose}>
            <NoIcon />
          </IconButton>
        </ButtonBlock>
      </ConfirmDelete>
    );
  }
);

export default ConfirmSendComponent;
