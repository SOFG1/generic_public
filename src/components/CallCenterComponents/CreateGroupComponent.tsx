import React, { useCallback, useMemo, useState } from "react";
import styled from "styled-components";
import { handle } from "../../api";
import { GoogleAds } from "../../api/googleAds";
import { useAppActions } from "../../store/app";
import { useGoogleAdsState } from "../../store/googleAds";
import { useUserState } from "../../store/user";
import { desktopBp } from "../../styles/variables";
import { Button } from "../../UI/Button";
import { Dropdown } from "../../UI/Dropdown";
import { Input } from "../../UI/Input";
import { HintText, StyledHint } from "../../UI/InputHint/InputHint";
import { Loader } from "../../UI/Spinners";
import { InfoIcon } from "../../UI/Svg";
import { Title } from "../common/Title";
import { activityList } from "../../config/userActivityList";
import { useTranslation } from "react-i18next";

const StyledTitle = styled(Title)`
  margin-top: 0;
`;

const InputsContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1.04vw;
  padding: 0 7.81vw 0.52vw;
  @media screen and (max-width: ${desktopBp}) {
    gap: 13px;
    padding: 0 98px 7px;
  }
  @media screen and (max-width: 600px) {
    padding: 10px;
  }
`;

const InputBox = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 0.52vw;
  @media screen and (max-width: ${desktopBp}) {
    gap: 7px;
  }
`;

const ClearButton = styled(Button)`
  padding: 4px 5px;
  font-size: 0.73vw;
  width: 4.69vw;
  align-self: flex-end;
  @media screen and (max-width: ${desktopBp}) {
    font-size: 10px;
    width: 59px;
  }
`;

const StyledAddBtn = styled(Button)`
  margin: 0 auto 1.04vw;
  @media screen and (max-width: ${desktopBp}) {
    margin: 0 auto 13px;
  }
`;

const StyledLoader = styled(Loader)`
  margin: 0 auto;
`;

const CreateGroupComponent = React.memo(() => {
  const { t } = useTranslation()
  const { token } = useUserState();
  const { campaigns, selectedCustomerId } = useGoogleAdsState();
  const { onShowAlert } = useAppActions();
  const [selectedCampaignId, setSelectedCampaignId] = useState<string>("");
  const [groupName, setGroupName] = useState<string>("");
  const [isFetching, setIsFetching] = useState<boolean>(false);

  const handleAdd = useCallback(async () => {
    const reqData = {
      customer_id: String(selectedCustomerId),
      campaign_id: selectedCampaignId,
      ad_group_name: groupName,
    };
    if (token) {
      setIsFetching(true);
      const [dataRes, dataErr] = await handle(
        GoogleAds.createGroup(token, reqData)
      );
      setIsFetching(false);
      if (dataRes) {
        onShowAlert(true, "Success");
      }
      if (dataErr) {
        onShowAlert(false, dataErr.error);
        console.log(dataErr);
      }
    }
  }, [token, selectedCustomerId, selectedCampaignId, groupName]);

  const showClearButton = useMemo(() => {
    return selectedCampaignId || groupName;
  }, [selectedCampaignId, groupName]);

  const handleClear = () => {
    setSelectedCampaignId("");
    setGroupName("");
  };

  const campaignsOptions = useMemo(() => {
    return campaigns.map((c) => ({ item: c.name, value: c.campaign_id }));
  }, [campaigns]);

  return (
    <>
      <StyledTitle className = "styled_create_campaign_label">Create group</StyledTitle>
      <InputsContainer className = "styled_create_campaign_inputs">
        <InputBox>
          <Dropdown
            placeholder="Campaign"
            label="Campaign"
            value={selectedCampaignId}
            onSelect={setSelectedCampaignId}
            options={campaignsOptions}
          />
          <StyledHint>
            <HintText>{t("google_ads-create_group-h1")}</HintText>
            <InfoIcon />
          </StyledHint>
        </InputBox>
        <InputBox>
          <Input
            name="ad_group_name"
            type="text"
            placeholder="ad_group_name"
            label="ad_group_name"
            value={groupName}
            onChange={setGroupName}
          />
          <StyledHint>
            <HintText>{t("google_ads-create_group-h2")}</HintText>
            <InfoIcon />
          </StyledHint>
        </InputBox>
        {showClearButton && (
          <ClearButton onClick={handleClear} data-action={activityList["call-center-google-group_clear"]}>Clear all</ClearButton>
        )}
      </InputsContainer>
      {isFetching ? (
        <StyledLoader />
      ) : (
        <StyledAddBtn onClick={handleAdd} data-action={activityList["call-center-google-group_create"]}>Add a group</StyledAddBtn>
      )}
    </>
  );
});

export default CreateGroupComponent;
