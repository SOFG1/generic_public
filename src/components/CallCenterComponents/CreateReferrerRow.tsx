import React, {useCallback, useState} from "react";
import {Button} from "../../UI/Button";
import {useTranslation} from "react-i18next";
import {ITableColumn} from "../../store/rawData";
import {useCallCenterState} from "../../store/callCenter";
import {CreateRowComponent} from "../RawDataComponents";

const CreateReferrerRow = React.memo(() => {
  const { t } = useTranslation();
  const { interviewees } = useCallCenterState();
  const [creatingRowData, setCreatingRowData] = useState<ITableColumn | null>(
    null
  );

  const handleCreate = useCallback(() => {
    setCreatingRowData({ referal: interviewees?.id || "" });
  }, []);

  return (
    <>
      <Button onClick={handleCreate}>{t("referrer_create")}</Button>
      <CreateRowComponent
        initialData={creatingRowData}
        onClose={() => setCreatingRowData(null)}
      />
    </>
  );
});

export default CreateReferrerRow;
