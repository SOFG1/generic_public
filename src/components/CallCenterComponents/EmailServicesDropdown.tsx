import React, { useCallback, useEffect, useState } from "react"
import styled from "styled-components"
import { DropdownWithSearch } from "../../UI/Dropdown";
import { desktopBp } from "../../styles/variables";
import { useTranslation } from "react-i18next";
import { useUserState } from "../../store/user";
import { handle } from "../../api";
import { CallCenter } from "../../api/callCenter";


const DropdownStyled = styled(DropdownWithSearch)`
  width: 11.98vw;
  @media screen and (max-width: ${desktopBp}) {
    width: 150px;
  }
  @media screen and (max-width: 850px) {
    width: 150px;
  }
  @media screen and (max-width: 400px) {
    width: 100%;
  }
`;


interface IProps {
    value?: number
    onChange: (v: number) => void
}


const EmailServicesDropdown = React.memo(({ value, onChange }: IProps) => {
    const { t } = useTranslation()
    const { token } = useUserState()
    const [options, setOptions] = useState<{ item: string, value: number }[]>([])

    const fetchServicesOptions = useCallback(async () => {
        if (token) {
            const [dataRes, dataErr] = await handle(CallCenter.getEmailServices(token))
            if (dataRes) {
                console.log(dataRes)
                const opts = dataRes.map((s: any) => ({ item: s.view_name, value: s.id }))
                setOptions(opts)
            }
            if (dataErr) {
                console.log(dataErr)
            }
        }
    }, [token])

    useEffect(() => {
        fetchServicesOptions()
    }, [fetchServicesOptions])



    return <DropdownStyled
        value={value || ""}
        label={t("call-center_email-service")}
        placeholder={t("call-center_email-service")}
        onSelect={onChange}
        options={options}
    />
})

export default EmailServicesDropdown