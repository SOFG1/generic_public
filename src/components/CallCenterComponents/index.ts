export {default as InfoComponent} from './InfoComponent' 
export {default as FbSelect} from './facebookSelect' 
export {default as GoogleCampaignsComponent} from './GoogleCampaignsComponent' 
export {default as GoogleGroupsComponent} from './GoogleGroupsComponent' 
export {default as GoogleAdsComponent} from './GoogleAdsComponent' 
export {default as StringListComponent} from './StringListComponent'
export {default as AudienceCampaignComponent} from './AudienceCampaignComponent'
export {default as CreateCampaignComponent} from './CreateCampaignComponent'
export {default as CampaignCriteriaComponent} from './CampaignCriteriaComponent'
export {default as CreateGroupComponent} from './CreateGroupComponent'
export {default as GroupCriteriaComponent} from './GroupCriteriaComponent'
export {default as FBFilterAudience} from './FBFilterAudience'
export {default as FBPageAudience} from './FBPageAudience'
export {default as FBCampaignAudeince} from './FBCampaignAudeince'
export {default as QuestionComponent} from './QuestionComponent'
export {default as UpdateReferrersModal} from './UpdateReferrersModal'
export {default as UpdateReferrersComponent} from './UpdateReferrersComponent'
export {default as CreateReferrerRow} from './CreateReferrerRow'
export {default as SmsCampaignComponent} from './SmsCampaignComponent'
export {default as SmsServicesDropdown} from './SmsServicesDropdown'
export {default as EmailServicesDropdown} from './EmailServicesDropdown'