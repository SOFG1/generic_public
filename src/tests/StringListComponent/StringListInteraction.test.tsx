import { render, screen, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { StringListComponent } from "../../components/CallCenterComponents";


describe("String list component interaction testing", () => {
  let stringArray: string[];
  let buttons: any[];
  const onChange = (values: string[]) => {
    stringArray = values;
  }; 

  beforeEach(() => {
    stringArray = ["123", "test"]; 
    render(<StringListComponent onChange={onChange} values={stringArray} />);
    buttons = screen.getAllByRole("button");
  }); 

  test("Add a new item in the string list", () => {
    //interaction
    fireEvent.click(buttons[2]);
    //result
    expect(stringArray).toEqual(["123", "test", ""]);
  });

  test("Remove last item in the string list", () => {
    //interaction
    fireEvent.click(buttons[1]);
    //result
    expect(stringArray).toEqual(["123"]);
  });

  test("Remove first item in the string list", () => {
    //interaction
    fireEvent.click(buttons[0]);
    //result
    expect(stringArray).toEqual(["test"]);
  });

  test("Change first string in the list", () => {
    //state
    const firstInput = screen.getAllByRole('textbox')[0]

    //interaction (set value to 'newValue')
    fireEvent.change(firstInput, {target: {value: 'newValue'}})

    //result
    expect(stringArray).toEqual(["newValue", "test"])
  })



  //Snapshot testing
  it('StringListComponent snapshot test', () => {

    const values = ['first', 'second', 'third', 'fourth', 'fifth', 'sixth']
  
      const tree = renderer
        .create(<StringListComponent onChange={(v) => {}} values={values} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });


});
