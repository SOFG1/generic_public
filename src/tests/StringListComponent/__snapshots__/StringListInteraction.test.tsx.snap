// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String list component interaction testing StringListComponent snapshot test 1`] = `
<div
  className="sc-tsFYE ha-DNZi"
>
  <div
    className="sc-bhVIhj gcFkvP"
  >
    <div
      className="sc-llJcti bfmfft sc-eGAhfa ljmKPN input input0"
    >
      <input
        className="sc-gicCDI gTEsHJ"
        dir="auto"
        name="0"
        onChange={[Function]}
        placeholder=""
        type="text"
        value="first"
      />
      <label
        className="sc-iIPllB kXojUT"
        dir="auto"
      />
    </div>
    <button
      className="sc-kYWVYA itLWnQ"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 7L10 10M13 13L10 10M10 10L13 7M10 10L7 13"
          stroke="#000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
        />
        <rect
          height="19"
          rx="9.5"
          stroke="#000"
          width="19"
          x="0.5"
          y="0.5"
        />
      </svg>
    </button>
  </div>
  <div
    className="sc-bhVIhj gcFkvP"
  >
    <div
      className="sc-llJcti bfmfft sc-eGAhfa ljmKPN input input1"
    >
      <input
        className="sc-gicCDI gTEsHJ"
        dir="auto"
        name="1"
        onChange={[Function]}
        placeholder=""
        type="text"
        value="second"
      />
      <label
        className="sc-iIPllB kXojUT"
        dir="auto"
      />
    </div>
    <button
      className="sc-kYWVYA itLWnQ"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 7L10 10M13 13L10 10M10 10L13 7M10 10L7 13"
          stroke="#000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
        />
        <rect
          height="19"
          rx="9.5"
          stroke="#000"
          width="19"
          x="0.5"
          y="0.5"
        />
      </svg>
    </button>
  </div>
  <div
    className="sc-bhVIhj gcFkvP"
  >
    <div
      className="sc-llJcti bfmfft sc-eGAhfa ljmKPN input input2"
    >
      <input
        className="sc-gicCDI gTEsHJ"
        dir="auto"
        name="2"
        onChange={[Function]}
        placeholder=""
        type="text"
        value="third"
      />
      <label
        className="sc-iIPllB kXojUT"
        dir="auto"
      />
    </div>
    <button
      className="sc-kYWVYA itLWnQ"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 7L10 10M13 13L10 10M10 10L13 7M10 10L7 13"
          stroke="#000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
        />
        <rect
          height="19"
          rx="9.5"
          stroke="#000"
          width="19"
          x="0.5"
          y="0.5"
        />
      </svg>
    </button>
  </div>
  <div
    className="sc-bhVIhj gcFkvP"
  >
    <div
      className="sc-llJcti bfmfft sc-eGAhfa ljmKPN input input3"
    >
      <input
        className="sc-gicCDI gTEsHJ"
        dir="auto"
        name="3"
        onChange={[Function]}
        placeholder=""
        type="text"
        value="fourth"
      />
      <label
        className="sc-iIPllB kXojUT"
        dir="auto"
      />
    </div>
    <button
      className="sc-kYWVYA itLWnQ"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 7L10 10M13 13L10 10M10 10L13 7M10 10L7 13"
          stroke="#000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
        />
        <rect
          height="19"
          rx="9.5"
          stroke="#000"
          width="19"
          x="0.5"
          y="0.5"
        />
      </svg>
    </button>
  </div>
  <div
    className="sc-bhVIhj gcFkvP"
  >
    <div
      className="sc-llJcti bfmfft sc-eGAhfa ljmKPN input input4"
    >
      <input
        className="sc-gicCDI gTEsHJ"
        dir="auto"
        name="4"
        onChange={[Function]}
        placeholder=""
        type="text"
        value="fifth"
      />
      <label
        className="sc-iIPllB kXojUT"
        dir="auto"
      />
    </div>
    <button
      className="sc-kYWVYA itLWnQ"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 7L10 10M13 13L10 10M10 10L13 7M10 10L7 13"
          stroke="#000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
        />
        <rect
          height="19"
          rx="9.5"
          stroke="#000"
          width="19"
          x="0.5"
          y="0.5"
        />
      </svg>
    </button>
  </div>
  <div
    className="sc-bhVIhj gcFkvP"
  >
    <div
      className="sc-llJcti bfmfft sc-eGAhfa ljmKPN input input5"
    >
      <input
        className="sc-gicCDI gTEsHJ"
        dir="auto"
        name="5"
        onChange={[Function]}
        placeholder=""
        type="text"
        value="sixth"
      />
      <label
        className="sc-iIPllB kXojUT"
        dir="auto"
      />
    </div>
    <button
      className="sc-kYWVYA itLWnQ"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 7L10 10M13 13L10 10M10 10L13 7M10 10L7 13"
          stroke="#000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
        />
        <rect
          height="19"
          rx="9.5"
          stroke="#000"
          width="19"
          x="0.5"
          y="0.5"
        />
      </svg>
    </button>
  </div>
  <button
    className="sc-hAsxaJ fHSKKA"
    onClick={[Function]}
  />
</div>
`;
